src/odemis/odemisd/mdupdater.py

```diff
@@ -298,25 +305,106 @@ def updateLightPower(power, light=light, comp_affected=comp_affected):
 
         return True
 
-    def observeSpectrograph(self, spectrograph, comp_affected):
+    def getMonochromatorBandwidth(self, spectrograph: Optional[model.HwComponent]) -> Optional[Tuple[float, float]]:
+        """
+        Get the bandwidth of the monochromator based on the spectrograph settings.
+        :param spectrograph: a spectrograph component
+        :return: the bandwidth of the monochromator (min, max) in m, or None if not available
+        """
+        if spectrograph is None:
+            return None
+
+        # For monochomators, we need to know the minimum and maximum wavelength
+        # detected based on the spectrograph settings. => Update MD_OUT_WL
+        pos = spectrograph.position.value
+        if 'slit-monochromator' not in pos:
+            logging.info("No 'slit-monochromator' axis was found, will not be able to compute monochromator bandwidth.")
+            wl = pos["wavelength"]
+            if wl < 10e-9:
+                return None
+            return (wl, wl)
+        else:
+            width = pos['slit-monochromator']
+            bandwidth = spectrograph.getOpeningToWavelength(width)  # always a tuple
+            return bandwidth
+
+    def getFilterBandwidth(self, filter: Optional[model.HwComponent]) -> Union[Iterable[float], str, None]:
+        """
+        Get the bandwidth of the filter based on the filter settings.
+        :param filter: a (light) "filter" (wheel) component (should have a "band" axis)
+        :return: the bandwidth of the filter (min, max) in m,
+                 or a string if it's defined just as a name,
+                 or None if there is no filter.
+        """
+        if filter is None:
+            return None
+
+        pos = filter.position.value
+        bandwidth = filter.axes["band"].choices[pos["band"]]
+        if bandwidth == model.BAND_PASS_THROUGH:
+            return None
+        elif isinstance(bandwidth, str):  # Sometimes the filter is just a name like "red".
+            return bandwidth
+        # The 3rd case is a tuple of floats
+        elif not isinstance(bandwidth, (tuple, list)) and all(isinstance(v, float) for v in bandwidth):
```

Comment:
I'm confused about when it would hit this `elif`, is that if bandwidth is a dict and all keys in the dict are floats? Or if bandwidth is a set? Or am I not reading it correctly?